content/blog/deploying-the-infrastructure-of-oauth-server-for-cms-app/index.md

```diff
@@ -0,0 +1,363 @@
+For the Authorization callback URL we should enter `https://{{the domain of your OAuth App}}/callback/github` which is specified by the main.go.
+
+```go
+    github.New(
+        os.Getenv("GITHUB_KEY"), os.Getenv("GITHUB_SECRET"),
+        // concatenate with the host name
+        fmt.Sprintf("%s/callback/github", host),
+        "public_repo",
+    )
+```
+
+If you are using different storaging system like Bitbucket and Gitlab, you can specify callback url to be `https://{{the domain of your OAuth App}}/callback/{{ backend name }}` and relace `{{backend name}}` with `bitbucket` or `github` correspondingly.
+
+Then we can copy and paste the github key and secret as stack configuration as well.
+
+```bash
+$ pulumi config set netlify-cms-oauth-provider-infrastructure:githubKey {{YOUR_GITHUB_KEY}}
+$ pulumi config set --secret netlify-cms-oauth-provider-infrastructure:githubSecret
+$ {{YOUR_GITHUB_SECRET}}
+```
+
+`--secret` would safely encrypt your github secret. Also,
+don't directly append the secret to the command like this `$ pulumi config set --secret netlify-cms-oauth-provider-infrastructure:githubKey {{YOUR_GITHUB_SECRET}}`.
+Because it might cause the secret to being stored inside the command memory. We should only specify the key without a name and hit ENTER key, then you can type secret on next line(won't show the value)
+
+For `TARGET_PORT` that is how we passed in the `targetGroupPort` configuration value to main.go. That is also the port number that we want the server itself to serve. We also checked if this environment variable is set in main.go in case we need to test main.go locally without Pulumi.
+
+```go
+targetGroupPort := os.Getenv("TARGET_PORT")
+    if targetGroupPort == "" {
+        targetGroupPort = "80"
+    }
+    listenPort := ":" + targetGroupPort
+    fmt.Print("Started running on", listenPort, "\n")
+    // listen on port 80 where we created the target group
+    fmt.Println(http.ListenAndServe(listenPort, nil))
+```
+
+As mentioned previously, `GITHUB_SCOPE` is the environment variable to tell main.go what kind of access CMS needs to have on the target repository. It is also an optional stack configuration where the default is set as public_repo in the main.go. We can specify multiple scopes by a separate scope with commas.
+
+### Create a Certificate for the OAuth Server
+
+The process is similar how we created certificate for CMS in our [previous post]({{< relref "/blog/deploying-netlify-cms-on-aws" >}}).
+Similarly, we also referenced Pulumi's [static website example](https://github.com/pulumi/examples/tree/master/aws-ts-static-website).
+
+We can extract what we need just for creating the certificate and delete irrelevant parts including uploading content to the S3 bucket and configure CloudFront. Then the rest only involves creating an east region provider, a certificate, and certificate validation.
```

Comment:
```suggestion
We use just what we need just for creating the certificate.
```

Reply:
I think I should still point out which part of the example that is irrelevant/ relevant and show the way to use Pulumi examples

Reply:
@zephyrz73 I do see your point about wanting to highlight that we can easily just take the relevant parts from the other infrastructure app. :) However, I agree that the suggestion made by @spara captures the same thing you are saying but in a succinct way and does imply that we are taking what we need just for creating the certificate itself. I recommend applying the suggestion here.